core/client/src/client.rs

```diff
@@ -579,6 +594,74 @@ impl<B, E, Block> Client<B, E, Block> where
 		let info = self.backend.blockchain().info().map_err(|e| error::Error::from_blockchain(Box::new(e)))?;
 		Ok(self.header(&BlockId::Hash(info.best_hash))?.expect("Best block header must always exist"))
 	}
+
+	/// Get the most recent block hash of the best (longest) chains
+	/// that contain block with the given `target_hash`.
+	/// If `maybe_max_block_number` is `Some(max_block_number)`
+	/// the operation is performed as if there are no blocks greater `max_block_number`
+	/// ignoring all blocks with numbers greater
+	/// TODO [snd] possibly implement this on blockchain::Backend and just redirect here
+	pub fn best_containing(&self, target_hash: Block::Hash, maybe_max_number: Option<<<Block as BlockT>::Header as HeaderT>::Number>) -> error::Result<Option<Block::Hash>> {
```

Comment:
I think there's a `NumberFor<Block>` type definition in the `runtime_primitives` that might be useful.

Reply:
simplified